### PR TITLE
Update transaction structure

### PR DESCRIPTION
**Pull request by IViol**
- add `hasInput` field into `Transaction` structure. It is needed to define "abstract" contract method call.
- remove redundant `isCanceled` flag from `Transaction` structure.
- add "filtering" flags into `TransactionsState`.
- fix validation of transactions from block explorer. Add ability to show "abstract" contract method calls.
- fix **critical** bug connected with incorrect `nonce` validation from web3 payload, that caused infinite ETH node syncing.
- update blocks & transactions selectors. Add ability to omit some optional parameters, that always defined in the app, such as network & currently active wallet address.
- update `selectTransactionById` / `selectPendingTransactionByHash`. These selectors was made flexible, added ability to search items with different precision (passing different set of arguments)

### CODE REVIEW

File: flow-typed/redux/transactions.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ declare type TransactionFromBlockExplorer = {|
   +hash: Hash,
   +to: string,
   +from: string,
+  +input: string,
```
Comment (flexxnn) [inline flow-typed/redux/transactions.js]: I think we need to leave comment here. When this PR will be closed nobody will know what is this

File: src/services/blockExplorer.js [hunk 2]
```diff
@@ -86,34 +81,37 @@ function checkETHTransaction(data: Object): boolean {
     type.isString(data.gas) &&
     type.isString(data.from) &&
     type.isString(data.hash) &&
+    type.isString(data.input) &&
+    type.isString(data.nonce) &&
```
Comment (flexxnn) [inline src/services/blockExplorer.js]: As I remember, that getNonce method returns number. Please check that this is always a string

File: src/services/blockExplorer.js [hunk 3]
```diff
@@ -86,34 +81,37 @@ function checkETHTransaction(data: Object): boolean {
...
 
     const {
+      input,
       value,
       isError,
       contractAddress,
     }: Object = item
 
     const isEmptyAmount: boolean = isZero(value)
+    const isEmptyInput: boolean = (input === '0x')
```
Comment (flexxnn) [inline src/services/blockExplorer.js]: `input === '0x' || input === ''` ?

File: src/services/blockExplorer.js (hunk 2 above)
Comment (IViol) [reply src/services/blockExplorer.js]: I checked, etherscan API returns all fields as strings

File: src/services/blockExplorer.js (hunk 3 above)
Comment (IViol) [reply src/services/blockExplorer.js]: I checked, etherscan API returns `0x` string as empty input

File: flow-typed/redux/transactions.js (hunk 1 above)
Comment (IViol) [reply flow-typed/redux/transactions.js]: This field's name does not differ from ETH node getTransaction response format.
But, ok, will add some note in next PR

**Pull request merged**